cdap-app-templates/cdap-program-report/src/main/scala/co/cask/cdap/report/ReportGenerationHelper.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.report
+
+import co.cask.cdap.report.proto.ReportGenerationRequest
+import co.cask.cdap.report.util.Constants
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.{SQLContext, SparkSession}
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConversions._
+
+class ReportGenerationHelper(private val spark: SparkSession) {
+
+  def this(sc: SparkContext) {
+    this(new SQLContext(sc).sparkSession)
+  }
+
+  def generateReport(request: ReportGenerationRequest, inputPaths: java.util.List[String], outputPath: String): Long = {
+    import ReportGenerationHelper._
+    import spark.implicits._
+    val df = spark.read.format("com.databricks.spark.avro").load(asScalaBuffer(inputPaths): _*)
```

Comment:
No need to have `asScalaBuffer`. Just add an `import scala.collection.JavaConversions._`